parser/lexer.go

```diff
@@ -294,8 +294,18 @@ func startWithSharp(s *Scanner) (tok int, pos Pos, lit string) {
 func startWithDash(s *Scanner) (tok int, pos Pos, lit string) {
 	pos = s.r.pos()
 	if !strings.HasPrefix(s.r.s[pos.Offset:], "-- ") {
-		tok = int('-')
-		s.r.inc()
+		if strings.HasPrefix(s.r.s[pos.Offset:], "->") {
```

Comment:
`if` nested too much, you'd better reverse the logic:

```
if strings.HasPrefix(s.r.s[pos.Offset:], "-- ") {
    // skip comment and read next token
    // return
}
if strings.HasPrefix(s.r.s[pos.Offset:], "->") {
    // this branch test -> and ->>
    // return
}
tok = int('-')
s.r.inc()
return
```

Special cases goes first, and returns as soon as possible, followed by normal case.
This way the code can be cleaner.